calculateSNR.m

```diff
@@ -286,86 +305,88 @@
                    :(cfg.binSize + 1 - 1 - cfg.gap) ...
                    (cfg.binSize + 1 + 1 + cfg.gap) ...
                    :(cfg.binSize + 1 + cfg.binSize / 2 + cfg.gap)];
 
+        % extract magnitudes of signal and noise from the average segment 
         AmpNoise = mXavgHarmonics(noiseFs, :);
         NoiseMean = mean(AmpNoise, 1);
         NoiseSD = std(AmpNoise, 0, 1);
+        % get zscore 
         targetHarmonicsZ = (mXavgHarmonics(cfg.binSize + 1, :) - NoiseMean) ./ NoiseSD;
 
         % save map as nii
         newFileName = ['AvgZHarmonics_', boldFileName, '.nii'];
-        writeMap(targetHarmonicsZ, maskHdr, maskImg, newFileName, destinationDir);
-        %     writeMap(targetHarmonicsZ, maskHdr.fname, newFileName, destinationDir);
-
-        % plot best voxels
-        [~, idxSorted] = sort(targetHarmonicsZ, 'descend');
-        idxSorted(isnan(targetHarmonicsZ(idxSorted))) = [];
-        nMax = 10;
-
-        f = figure('color', 'white', 'Position', [131 728 1744 140]);
-        pnl = panel(f);
-        pnl.pack('h', nMax);
-
-        for iVox = 1:nMax
-            mXbest = mXavgHarmonics(:, idxSorted(iVox));
-            pnl(iVox).select();
-            h = stem(mXbest, ...
-                     'marker', 'none', ...
-                     'color', [0.6, 0.6, 0.6], ...
-                     'linew', 4);
-            hold on;
-            h = stem(cfg.binSize + 1, mXbest(cfg.binSize + 1), ...
-                     'marker', 'none', ...
-                     'color', 'r', ...
-                     'linew', 4);
-            title(sprintf('z=%.2f  vox=%d', targetHarmonicsZ(idxSorted(iVox)), ...
-                          idxSorted(iVox)));
-            set(gca, 'xtick', []);
-        end
-
-        pnl.ylabel('amplitude');
-        pnl.marginbottom = 3;
+        zHarmonicsImg = writeMap(targetHarmonicsZ, maskHdr, maskImg, newFileName, destinationDir);
 
+        coordHarmonics = getVoxelCoordinate(boldHdr, zHarmonicsImg, maskImg, voxelNbToPlot);
+        
+        % plot averge segment from the spectra for best voxels
+        % for a clear representation, does it make sense to use fixed scale
+        % across voxels?
+        f = plotAvgHarmBestVox(mXavgHarmonics, coordHarmonics); 
+        
         % save figure
         newFileName = ['AvgZHarmonics-bestVoxMean_', boldFileName, '.fig'];
         saveas(f, fullfile(destinationDir, newFileName));
         close(f);
 
-        % plot best voxels and save figure
-        f = plotmXBestVox(freq, mXSNRAmp, targetHarmonicsZ, 10, cfg.idxHarmonics);
+        % plot whole spectra for best voxels and save figure
+        % the coordinate of best harmonics differ from coord of highest SNR
+        % on target freuency - for Fig1 differs from Fig3
+        f = plotmXBestVox(freq, mXbl, coordHarmonics, cfg.idxHarmonics);
         newFileName = ['AvgZHarmonics-bestVox_', boldFileName, '.fig'];
         saveas(f, fullfile(destinationDir, newFileName));
         close(f);
```

Comment:
this is also my note. the coordinates of best harmonics will differ from the coord of best target frequency. then would we present 3 figures? are they all informative?